go/internal/services/keys/status.go

```diff
@@ -0,0 +1,160 @@
+		return nil
+	}
+
+	switch k.Status {
+	case StatusValid:
+		return nil
+	case StatusNotFound:
+		return fault.New("key does not exist",
+			fault.Code(codes.Auth.Authentication.KeyNotFound.URN()),
+			fault.Internal("key does not exist"),
+			fault.Public("We could not find the requested key."),
+		)
+	case StatusDisabled:
+		message := k.message
+		if message == "" {
+			message = "the key is disabled"
+		}
+		return fault.New("key is disabled",
+			fault.Code(codes.Auth.Authorization.KeyDisabled.URN()),
+			fault.Internal(message),
+			fault.Public("The key is disabled."),
+		)
+	case StatusExpired:
+		message := k.message
+		if message == "" {
+			message = "the key has expired"
+		}
+		return fault.New("key has expired",
+			fault.Code(codes.Auth.Authorization.Forbidden.URN()),
+			fault.Internal(message),
+			fault.Public(message),
+		)
+	case StatusWorkspaceDisabled:
+		return fault.New("workspace is disabled",
+			fault.Code(codes.Auth.Authorization.WorkspaceDisabled.URN()),
+			fault.Internal("workspace disabled"),
+			fault.Public("The workspace is disabled."),
+		)
+	case StatusWorkspaceNotFound:
+		return fault.New("workspace not found",
+			fault.Code(codes.Data.Workspace.NotFound.URN()),
+			fault.Internal("workspace disabled"),
+			fault.Public("The requested workspace does not exist."),
+		)
+	case StatusForbidden:
+		message := k.message
+		if message == "" {
+			message = "Forbidden"
+		}
+		return fault.New("forbidden",
+			fault.Code(codes.Auth.Authorization.Forbidden.URN()),
+			fault.Internal(message),
+			fault.Public(message),
+		)
+	case StatusInsufficientPermissions:
+		message := k.message
+		if message == "" {
+			message = "Insufficient permissions to access this resource."
+		}
+		return fault.New("insufficient permissions",
+			fault.Code(codes.Auth.Authorization.InsufficientPermissions.URN()),
+			fault.Internal(message),
+			fault.Public(message),
+		)
+	case StatusUsageExceeded:
+		message := k.message
+		if message == "" {
+			message = "Key usage limit exceeded."
+		}
+		return fault.New("key usage limit exceeded",
+			fault.Code(codes.Auth.Authorization.Forbidden.URN()),
+			fault.Internal(message),
+			fault.Public(message),
+		)
+	case StatusRateLimited:
+		message := k.message
+		if message == "" {
+			message = "Rate limit exceeded"
+		}
+		return fault.New("rate limit exceeded",
+			fault.Code(codes.Auth.Authorization.Forbidden.URN()),
+			fault.Internal(message),
+			fault.Public(message),
+		)
+	default:
+		return fault.New("key verification failed",
+			fault.Code(codes.Auth.Authorization.Forbidden.URN()),
+			fault.Internal("key verification failed with unknown status"),
+			fault.Public("Key verification failed."),
+		)
+	}
+}
+
+// ToOpenAPIStatus converts our internal KeyStatus to the OpenAPI response status type.
+// This mapping ensures consistency between internal validation and external API responses.
+func (k *KeyVerifier) ToOpenAPIStatus() openapi.KeysVerifyKeyResponseDataCode {
```

Comment:
I wonder if we should use our `go/pkg/codes` codes for these.. maintaining separate codes for different things will become annoying.

maybe something like `err:unkey:authorization:key_disabled`. There's probably a better fitting one or we create a new category, but that would be the idea.